modules/fbs/includes/fbs.availability.inc

```diff
@@ -94,11 +140,7 @@ function fbs_availability_holdings($provider_ids) {
         $result_holding['placement'][] = $material_description[0]->materialGroupName;
       }
 
-      module_load_include('client.inc', 'opensearch');
-      $entity_id = ding_provider_build_entity_id($provider_ids[0]);
-      $suffix_type = variable_get('fbs_holdings_suffix_type', 'shelf_mark');
```

Comment:
Moving this *out* of the loop started on L75 seems to be a key change as `ding_provider_build_entity_id()` will invoke the search provider and [`opensearch` will issue a search](https://github.com/ding2/ding2/blob/078ff150143dd461b32ad842007716bf65328937/modules/opensearch/includes/opensearch.search.inc#L624).

This change seems unrelated to the rest of the changes here. Could you extract that into a separate PR?

Reply:
No, it's not moved outside the loop???

It checks if the entity id exists in the current loop and loads ding_provider_build_entity_id(reset($item->recordId))